File: Leetcode/May/L796_Rotate_String.java
package Leetcode.May;

public class L796_Rotate_String {
    public static void main(String[] args) {
        String s = "abcde", goal = "cdeab";
        System.out.println(rotateString(s, goal));
    }

    public static boolean rotateString(String s, String goal) {
        if (s.length() != goal.length()) return false;
        String str = s + s;
        return str.contains(goal);
    }
}

File: Striver_DSA_Sheet/DSA_in_Python/String/L796_Rotate_String.py
def rotateString(s: str, goal: str) -> bool:
    if len(s) != len(goal):
        return False
    for i in range(len(s)):
        if s == goal:
            return True
        s = rotateByOne(s)
    return False


def rotateByOne(s: str) -> str:
    s = list(s)
    char = s[-1]
    for i in range(len(s) - 1, 0, -1):
        s[i] = s[i - 1]
    s[0] = char
    return ''.join(s)

print(rotateString("abcde", "cdeab"))